backend/src/middleware/requireIntegrationAuth.ts

```diff
@@ -21,48 +22,40 @@ const requireIntegrationAuth = ({
 	return async (req: Request, res: Response, next: NextFunction) => {
 		// integration authorization middleware
 
-		try {
-			const { integrationId } = req.params;
+		const { integrationId } = req.params;
 
-			// validate integration accessibility
-			const integration = await Integration.findOne({
-				_id: integrationId
-			});
+		// validate integration accessibility
+		const integration = await Integration.findOne({
+			_id: integrationId
+		});
 
-			if (!integration) {
-				throw new Error('Failed to find integration');
-			}
-			
-			await validateMembership({
-				userId: req.user._id.toString(),
-				workspaceId: integration.workspace.toString(),
-				acceptedRoles,
-				acceptedStatuses
-			});
-
-			const integrationAuth = await IntegrationAuth.findOne({
-				_id: integration.integrationAuth
-			}).select(
-				'+refreshCiphertext +refreshIV +refreshTag +accessCiphertext +accessIV +accessTag +accessExpiresAt'
-			);
-
-			if (!integrationAuth) {
-				throw new Error('Failed to find integration authorization');
-			}
+		if (!integration) {
+			return next(UnauthorizedRequestError({message: 'Failed to locate Integration'}))
```

Comment:
Same inquiry as previous `Bot` inquiry regarding error type. Should we be using `UnauthorizedRequestError` here or should there be a more specific Error type that implies that resource could not be found?

Reply:
Same thing here as well. I answered in previous one.